src/components/ReportActionItemIOUAction.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../ONYXKEYS';
+import ReportActionItemIOUQuote from './ReportActionItemIOUQuote';
+import ReportActionPropTypes from '../pages/home/report/ReportActionPropTypes';
+import ReportActionItemIOUPreview from './ReportActionItemIOUPreview';
+import Navigation from '../libs/Navigation/Navigation';
+import ROUTES from '../ROUTES';
+
+const propTypes = {
+    // All the data of the action
+    action: PropTypes.shape(ReportActionPropTypes).isRequired,
+
+    // The associated chatReport
+    chatReportID: PropTypes.number.isRequired,
+
+    // Should render the preview Component?
+    shouldDisplayPreview: PropTypes.bool.isRequired,
+
+    /* --- Onyx Props --- */
+    // ChatReport associated with iouReport
+    chatReport: PropTypes.shape({
+        // The participants of this report
+        participants: PropTypes.arrayOf(PropTypes.string),
+    }),
+};
+
+const defaultProps = {
+    chatReport: {},
+};
+
+const ReportActionItemIOUAction = ({
+    action,
+    chatReportID,
+    shouldDisplayPreview,
+    chatReport,
+}) => (
+    <View>
+        <ReportActionItemIOUQuote
+            action={action}
+            shouldShowViewDetailsLink={chatReport.participants.length <= 2}
+            onViewDetailsPressed={() => Navigation.navigate(ROUTES.getIouDetailsRoute(
+                chatReportID, action.originalMessage.IOUReportID,
+            ))}
+        />
+        {shouldDisplayPreview && (
+            <ReportActionItemIOUPreview
+                iouReportID={action.originalMessage.IOUReportID}
+                onPayButtonPressed={() => Navigation.navigate(ROUTES.getIouDetailsRoute(
```

Comment:
This callback should be DRYed up (it can be declared at the top of this function and then passed as a reference).